etl/steps/data/garden/irena/2025-08-22/renewable_power_generation_costs.py

```diff
@@ -8,19 +8,53 @@
 # Unfortunately, they also dropped data for years prior to 2010, for various indicators, that were given in the previous release (in constant 2023$).
 # Therefore, in this step, I combine the old data with the latest release (prioritizing the latest release, where there is overlap).
 # To do that, I convert the old data (in constant 2023$) to constant 2024$.
 # IRENA costs are given in the latest year's USD, so we convert other costs to the same currency.
 LATEST_YEAR = 2024
 PREVIOUS_YEAR = LATEST_YEAR - 1
-# Convert costs to constant {LATEST_YEAR} USD, using the World Bank's GDP deflator:
-# https://data.worldbank.org/indicator/NY.GDP.DEFL.ZS.AD?end=2024&locations=US&start=1990&view=chart&year=2024
-# I downloaded the CSV file, and took the values for the last two years for the US, namely 122.27 (for 2023), and 125.23 (for 2024).
-# The resulting factor is the result of dividing the value for 2024 by the value for 2023:
-USDPREVIOUS_TO_USDLATEST = 1.024
+
+
+def adjust_old_data_for_inflation(tb_old, tb_deflator, tb):
+    # Use the GDP deflator (linked series) to convert from constant USD of PREVIOUS_YEAR to constant USD of LATEST_YEAR.
+    tb_deflator = (
+        tb_deflator[tb_deflator["year"].isin([PREVIOUS_YEAR, LATEST_YEAR])][["country", "year", "gdp_deflator_linked"]]
+        .pivot(index=["country"], columns=["year"], values="gdp_deflator_linked")
+        .reset_index()
+    )
+    tb_deflator = tb_deflator.dropna(how="all", subset=[PREVIOUS_YEAR, LATEST_YEAR]).reset_index(drop=True)
+    # Check that we have deflator data for PREVIOUS_YEAR for all countries.
+    assert set(tb_old["country"]) - set(tb_deflator[(tb_deflator[PREVIOUS_YEAR].notnull())]["country"]) == {"World"}
+    # Only one country will not be able to be adjusted, namely South Korea.
+    assert set(tb_old["country"]) - set(tb_deflator[(tb_deflator[LATEST_YEAR].notnull())]["country"]) == {
+        "World",
+        "South Korea",
+    }
+    # Calculate the adjustment factor.
+    tb_deflator["adjustment"] = tb_deflator[2024] / tb_deflator[2023]
+
+    # Add adjustment column to old table.
+    tb_old = tb_old.merge(tb_deflator[["country", "adjustment"]], on=["country"], how="left")
+
+    # Assign US adjustment to the World (for which we don't have an alternative).
+    tb_old.loc[tb_old["country"] == "World", "adjustment"] = tb_deflator[(tb_deflator["country"] == "United States")][
+        "adjustment"
+    ].item()
+
+    # Column by column in the old table, convert to constant USD of the latest year.
+    for column in tb_old.drop(columns=["country", "year", "adjustment"]).columns:
+        # Sanity check.
+        error = f"Unexpected units for column {column}."
+        assert tb[column].metadata.unit == f"constant {LATEST_YEAR} US$ per kilowatt-hour", error
+        # Fill adjustment factor with 1 for countries for which we can't do the adjustment (namely South Korea).
+        tb_old[column] *= tb_old["adjustment"].fillna(1)
```

Comment:
I am not sure if it is the best way to proceed here. The South Korean series will end up with a mix of 2023 and 2024 constant prices, which is hard to explain. Is it too bad to just drop the old series for them?

If you really want to proceed, you need to address that in `description_processing`

Reply:
Good point, I've just removed South Korea from the old table (as a result, we lose a few data points for onshore wind for South Korea).